### ISSUE TITLE
Octree Data Structure Support

### ISSUE THREAD

**Issue by aradhyamathur** [facebookresearch/pytorch3d#393]
## 🚀 Feature
Support for Octrees based neural networks 

## Motivation

Recently several research works by Wang et al.  have proposed Octree-Based CNNs for various shape analysis tasks. A similar feature implementation that allows the operations over octrees could be of great benefit especially due to the reduced memory footprint that octrees allow for 3D data. 

## Pitch
The current implementation exists in tensorflow only given [here](https://github.com/microsoft/O-CNN). It would great if pytorch3d too has similar features to allow the creation of octree based neural networks. Would be glad to contribute.

**Comment by gkioxari** [comment]
This sounds like a great suggestion! We will discuss this with our team and will comment here for a timeline.

**Comment by albertotono** [comment]
It seems that two days after this conversation they added support for pytorch https://github.com/microsoft/O-CNN/tree/master/pytorch

**Comment by aradhyamathur** [comment]
Yeah but I guess it will be better to have this data structure in pytorch3D as well, to allow some more flexibility to users to use octrees.

**Comment by albertotono** [comment]
They also confirmed that it is still under development.
